compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -988,6 +988,18 @@ class Namer { typer: Typer =>
          *  provided `mbr` is accessible and of the right implicit/non-implicit kind.
          */
         def addForwarder(alias: TermName, mbr: SingleDenotation, span: Span): Unit =
+          def adaptForwarderParams(acc: List[List[tpd.Tree]], tp: Type, prefss: List[List[tpd.Tree]])
+            : List[List[tpd.Tree]] = tp match
+              case mt: MethodType =>
+                if mt.paramInfos.nonEmpty && mt.paramInfos.last.isRepeatedParam then
+                  val prefsR = prefss.head.reverse
+                  val vararg = ctx.typeAssigner.seqToRepeated(prefsR.head)
+                  val prefsR1 = vararg :: prefsR.tail
+                  adaptForwarderParams(prefsR1.reverse :: acc, mt.resType, prefss.tail)
```

Comment:
Instead of reversing twice, using `:+` for destructing and constructing would be clearer I think, and not more expensive.

Reply:
I was thinking that :+ would be 3n (`.init`, `:+`, and `.last`), rather than 2x  `.reverse` but i guess its not that huge a list typically

Reply:
Maybe I'm misunderstanding but instead of .init and .last you can do `val a :+ b = xs` and that should only require do one traversal.

Reply:
wow I was not aware of that extractor, thanks :)